src/common/load-plugins.js

```diff
@@ -34,11 +33,16 @@ function loadPlugins(plugins, pluginSearchDirs) {
     let requirePath;
     try {
       // try local files
-      requirePath = resolve.sync(path.resolve(process.cwd(), pluginName));
-    } catch (e) {
+      requirePath = eval("require").resolve(
+        path.resolve(process.cwd(), pluginName)
+      );
+    } catch (_) {
       // try node modules
-      requirePath = resolve.sync(pluginName, { basedir: process.cwd() });
+      requirePath = eval("require").resolve(pluginName, {
+        paths: [process.cwd()]
+      });
     }
```

Comment:
I've merged #7517 into this one, to catch that error, remove this `try/catch` part and run 

```bash
yarn test tests_integration/__tests__/plugin-resolution.js
```